Java/PoolAndResourceFile/src/main/java/PoolAndResourceFile.java

```diff
@@ -0,0 +1,299 @@
+import java.io.*;
+import java.net.URISyntaxException;
+import java.security.InvalidKeyException;
+import java.time.Duration;
+import java.util.*;
+import java.util.concurrent.TimeoutException;
+
+import com.microsoft.azure.storage.*;
+import com.microsoft.azure.storage.blob.*;
+import org.apache.commons.io.IOUtils;
+
+import com.microsoft.azure.batch.*;
+import com.microsoft.azure.batch.auth.BatchSharedKeyCredentials;
+import com.microsoft.azure.batch.protocol.models.*;
+
+public class PoolAndResourceFile {
+
+    // Create IaaS pool if pool isn't exist
+    private static CloudPool CreatePool(BatchClient client, String poolId) throws BatchErrorException, IllegalArgumentException, IOException, InterruptedException, TimeoutException {
+        // Pool only have 1 A1 VM
+        String osPublisher = "OpenLogic";
+        String osOffer = "CentOS";
+        String poolVMSize = "STANDARD_A1";
+        int poolVMCount = 1;
+        long POOL_STEADY_TIMEOUT = 5 * 60 * 1000;
+        long VM_READY_TIMEOUT = 20 * 60 * 1000;
+
+        // Check if pool exists
+        if (!client.getPoolOperations().existsPool(poolId)) {
+
+            // See detail of creating IaaS pool at https://blogs.technet.microsoft.com/windowshpc/2016/03/29/introducing-linux-support-on-azure-batch/
+            // Get the sku image reference
+            List<NodeAgentSku> skus = client.getAccountOperations().listNodeAgentSkus();
+            String skuId = null;
+            ImageReference imageRef = null;
+
+            for (NodeAgentSku sku : skus) {
+                if (sku.getOsType() == OSType.LINUX) {
+                    for (ImageReference imgRef : sku.getVerifiedImageReferences()) {
+                        if (imgRef.getPublisher().equalsIgnoreCase(osPublisher) && imgRef.getOffer().equalsIgnoreCase(osOffer)) {
+                            imageRef = imgRef;
+                            skuId = sku.getId();
+                            break;
+                        }
+                    }
+                }
+            }
+
+            // Use IaaS VM with Linux
+            VirtualMachineConfiguration configuration = new VirtualMachineConfiguration();
+            configuration.setNodeAgentSKUId(skuId);
+            configuration.setImageReference(imageRef);
+
+            client.getPoolOperations().createPool(poolId, poolVMSize, configuration, poolVMCount);
+        }
+
+        long startTime = System.currentTimeMillis();
+        long elapsedTime = 0L;
+        boolean steady = false;
+
+        // Wait for a maximum of 5 minutes for the VM to be allocated
```

Comment:
Instead of saying "5 minutes" here, can just say "Wait for the VMs to be allocated"